### PR TITLE
Move Notebooks file association to a separate extension

### LINKED ISSUE #1850: LiveShare participants incorrectly opening every Clojure file as if via "Open with Notebook"
## TL;DR

In `v2.0.292` Clojure Notebooks were introduced.

When using Calva with LiveShare the participants cannot open any Clojure files as regular files; they always open as if they chose "Open with Notebook".

Downgrading to `v2.0.291` removes problem (running with VSCode `1.70.2`).

## Steps to reproduce:

* Host opens Clojure project
* Host starts LiveShare and shares link with participant
* Participant opens link, connects to Shared system
* Participant opens any Clojure file, either via:
  - cmd-p > find file > enter
  - cmd-p > find file > click
  - cmd-p > find file > click the icon for "open to the side"
* Participant sees the file opened as a Clojure Notebook, not a regular Clojure file

Furthermore, after some exploration we found that if the participant opens the context menu for a `.edn` file in the sidebar, we see:

vscode-edn

In comparison, a `.clj` file has the following context-menu:

vscode-clj

Notice the distinct lack of "Open With..." for the `.clj` file.  The host sees "Open With..." both for the `.edn` and `.clj` file.

In contrast, if the host opens a Clojure file and uses the "Focus Participants" icon, the file will open correctly as a Clojure file for the participant.

## Reducing false-positives

We have tried the following so far to reduce false-positives:
* Host disabled any other VSCode extension that may have added Notebook features, etc.
* Participant has VSCode with minimal extensions enabled
* Participant tried connecting via the VSCode LiveShare in a browser
* Sharing a project without a running REPL (so even without host connecting to any REPL, we still see participants opening files always as Notebooks)

## Candidates for additional tests (if necessary)

We have not yet tried:
* REPL via jack-in (we have a multi-project monorepo and use connect-to existing REPL). But considering the issue is visible without any connected REPL, I don't think this is a primary candidate.
* Running a sample Clojure project (we've only tried to reproduce on work projects for now)
* Participant is logged in to LiveShare system (participants have up to now been connecting as Guests; the host is logged in)
* Participant has some issues with multi-project monorepo and Calva gives a warning about `No project-uri was specified`. (I think this issue is independent, but throwing it out there just in case)

I am not sure if any of the steps that have not been tried are significant in reproducing the issue, but I mention them in case something rings a bell.

## Next Steps (short-term)

We have temporarily downgraded to `v2.0.291` to avoid this issue while pair programming via LiveShare.

- Are there any ways to disable the Notebook feature while still using the newest version of Calva?
- Can someone else reproduce and confirm this issue with LiveShare and Notebooks? Otherwise we will need to consider what other factors are contributing to this issue.

## Next steps (long-term)

Hopefully fix the bug :)

PS. Thanks for all the work with Calva! ❤️

### PR DESCRIPTION
**Pull request by PEZ**
Because of Calva Clojure Notebooks interfering with LiveShare sessions:

* https://github.com/MicrosoftDocs/live-share/issues/4765

I have moved the Notebook file association part of the Calva extension manifest over to a separate extension, [Calva Spritz](https://marketplace.visualstudio.com/items?itemName=betterthantomorrow.calva-spritz). It only provides these associations, and nothing else. All Notebook supporting code remains in Calva. Calva Spritz is downloaded and enabled together with Calva.  Once the LiveShare team fixes that bug, we can move the file associations back to Calva.

* Fixes #1850

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [ ] Tests
  - [x] Tested the particular change
  - [ ] Figured if the change might have some side effects and tested those as well.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
  - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

**Review by Cyrik — APPROVED** [review]
Seems like a great idea!

File: docs/site/live-share.md [hunk 1]
```diff
@@ -16,6 +16,9 @@ programming, for example.
 An extra nice thing is that each participant is using their own VSCode configuration, including fonts, colors,
 keyboard shortcuts, etc.
 
+!!! Note "Disable Calva Spritz to get rid of Notebooks interference"
+    The headline below **Calva Supports Live Share** is true. However, due to [a bug in LiveShare](https://github.com/MicrosoftDocs/live-share/issues/4765), Guest participants always get their Clojure files opened in the Calva [Clojure Notebooks](notebooks.md) editor. To workaround this issue Calva uses a ”side-car” extension named **Calva Spritz** for the Notebooks associations of Clojure files. You can disable that extension when participating in LiveShare sessions.
```
Comment (bpringe) [inline docs/site/live-share.md]:
```suggestion
    The headline below **Calva Supports Live Share** is true. However, due to [a bug in LiveShare](https://github.com/MicrosoftDocs/live-share/issues/4765), guest participants always get their Clojure files opened in the Calva [Clojure Notebooks](notebooks.md) editor. To workaround this issue Calva uses a ”side-car” extension named **Calva Spritz** for the Notebooks associations of Clojure files. You can disable that extension when participating in LiveShare sessions.
```

File: notebooks-spritz/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+A bubbly connection between Clojure file types and Notebooks in VS Code, effervescently crafted to support Calva users.
+
+Because of an issue with LiveShare in VS Code making all Notebook associated files open in Notebooks we have this side-car extension to go with Calva. It is a workaround until the issue is fixed by the LiveShare team.
+
+The issue is tracked here:
+
+* Calva issue: [LiveShare participants incorrectly opening every Clojure file as if via "Open with Notebook"](https://github.com/BetterThanTomorrow/calva/issues/1850)
+* LiveShare issue: [Guest opens Clojure file as a notebook (incorrectly)](https://github.com/MicrosoftDocs/live-share/issues/4765)
+
+This extension only provides the notebook file association of Clojure files. All notebook supporting code lives in Calva. As a Calva user you will get this extension automatically installed and enabled together with Calva. As long as you have it enabled you will have Clojure Notebooks in VS Code when using Calva. To disable Notebook association and make LiveShare sessions work better (without Notebooks, sadly).
```
Comment (bpringe) [inline notebooks-spritz/README.md]: Is this last sentence incomplete? Does it mean to mention disabling Spritz?

File: notebooks-spritz/spritz-icon.png
Comment (bpringe) [inline notebooks-spritz/spritz-icon.png:1]: Nice logo 😄

Comment (PEZ) [reply notebooks-spritz/spritz-icon.png:1]: 😄 Inspiration from here. 
image

**Pull request merged**